components/movementsensor/gpsrtk/gpsrtk.go

```diff
@@ -188,24 +190,29 @@ func (g *gpsrtk) connectToNTRIP() error {
 		return err
 	}
 
+	g.writer = bufio.NewWriter(g.correctionWriter)
+	g.reader, err = g.getStream()
```

Comment:
Note to myself: `g.reader` is always created by calling `g.getStream()`, which always returns a `TeeReader` based on either `g.vrsReaderWriter` (for VRS) or something else (for non-VRS). So, all my concerns about suddenly using `g.reader` instead of the older `g.readerWriter` are misplaced: it's the same data coming from the same place, just with an extra layer of indirection.

It's okay to have the `TeeReader` take different sources for VRS and non-VRS, because exactly 1 of those sources are used in each one, and there should never be a time when both are valid (there should never be a time when our correction data is simultaneously VRS and non-VRS).